projects/plugins/jetpack/modules/masterbar/admin-menu/class-wpcom-admin-menu.php

```diff
@@ -216,40 +221,56 @@ public function set_site_card_menu_class( array $menu ) {
 	}
 
 	/**
-	 * Returns the first available upsell nudge.
-	 *
-	 * @return array
+	 * Enqueue the script to async load the upsell nudge.
 	 */
-	public function get_upsell_nudge() {
-		require_lib( 'jetpack-jitm/jitm-engine' );
-		$jitm_engine = new JITM\Engine();
-
-		$message_path = 'calypso:sites:sidebar_notice';
-		$current_user = wp_get_current_user();
-		$user_id      = $current_user->ID;
-		$user_roles   = implode( ',', $current_user->roles );
-		$query_string = array(
-			'message_path' => $message_path,
-		);
-
-		// Get the top message only.
-		$message = $jitm_engine->get_top_messages( $message_path, $user_id, $user_roles, $query_string );
-
-		if ( isset( $message[0] ) ) {
-			$message = $message[0];
-			return array(
-				'content'                      => $message->content['message'],
-				'cta'                          => $message->CTA['message'], // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
-				'link'                         => $message->CTA['link'], // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
-				'tracks_impression_event_name' => $message->tracks['display']['name'],
-				'tracks_impression_cta_name'   => $message->tracks['display']['props']['cta_name'],
-				'tracks_click_event_name'      => $message->tracks['click']['name'],
-				'tracks_click_cta_name'        => $message->tracks['click']['props']['cta_name'],
-				'dismissible'                  => $message->is_dismissible,
-				'feature_class'                => $message->feature_class,
-				'id'                           => $message->id,
+	public function enqueue_async_upsell_nudge() {
+		if ( ! did_action( 'enqueue_block_editor_assets' ) ) {
+			$gutenberg_wpcom_path = ABSPATH . 'wp-content/plugins/gutenberg-wpcom/';
+
+			wp_register_script( 'jquery.wpcom-proxy-request', '/wp-content/js/jquery/jquery.wpcom-proxy-request.js', array( 'jquery' ), '20211111', false );
+			wp_enqueue_script(
+				'async-nudge-gutenberg-wpcom-apifetch',
+				plugins_url( 'gutenberg-wpcom-apifetch.js', $gutenberg_wpcom_path . 'gutenberg-wpcom-apifetch.js' ),
+				array(
+					'jquery.wpcom-proxy-request',
+					'lodash',
+					'wp-api',
+					'wp-api-fetch',
+					'wp-polyfill',
+					'wp-url',
+				),
+				filemtime( $gutenberg_wpcom_path . 'gutenberg-wpcom-apifetch.js' ),
+				false
 			);
+			wp_localize_script(
+				'async-nudge-gutenberg-wpcom-apifetch',
+				'wpcomGutenberg',
+				array(
+					'errorMessages' => array(
+						'apiFetchOffline' => __( 'You are probably offline.', 'jetpack' ),
+						'apiFetchUnknown' => __( 'An unknown error occurred.', 'jetpack' ),
+					),
+				)
+			);
+
+			$script_data = '_currentSiteId = ' . get_current_blog_id() . ';';
+		}
+
+		wp_enqueue_script( 'admin-menu-upsell-nudge', plugins_url( 'upsell-nudge.js', __FILE__ ), array(), $this->cache_bust, true );
+		wp_add_inline_script( 'admin-menu-upsell-nudge', $script_data, 'before' );
```

Comment:
I'd very much prefer `wp_localize_script()` to be able to use a more unique var name, however the existing scripts relies on an existing global `_currentSiteId` variable, so we're reusing that here.